### ISSUE TITLE
Document how to use CMakePackage in the Packaging Guide

### ISSUE THREAD

**Issue by citibeth** [LLNL/spack#2246]
What methods can I override, and why?

**Comment by citibeth** [comment]
@alalazo

**Comment by davydden** [comment]
Roughly, the most important is
```
def cmake_args(self):
       args = []
        ...
        return args
```
which should have all the xtra arguments to cmake apart from standard arguments (prefix, build type, and a few others). those are in a private-like function `_std_args` which is not to be messed around with.

Build type can be overriden in 
```
def build_type(self):
      return 'DebugRelease'
```

Finally for fancy cases one could tweak
```
def root_cmakelists_dir(self):
        """Directory where to find the root CMakeLists.txt"""
        return self.stage.source_path
```

That should be it.

**Comment by adamjstewart** [comment]
Until things are documented properly, you can view limited documentation by going to http://spack.readthedocs.io/en/latest/spack.html and searching for CmakePackage.

**Comment by alalazo** [comment]
Can this be closed as a subset of #2606 ?

**Comment by citibeth** [comment]
Addressed in #2780
